### PR TITLE
Add PureScript LSP/Highlighting support

### PR DESCRIPTION
**Pull request by ivanmoreau**
This PR adds basic support for the language using the `purescript-language-server`.

Release Notes:

- Added support for PureScript.

### CODE REVIEW

**Comment by ivanmoreau** [comment]
Screen Shot 2024-01-27 at 22 08 42

An example of the LSP/Highlighting properly working.

File: crates/zed/src/languages/purescript.rs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    fn process_diagnostics(&self, l: &mut lsp::PublishDiagnosticsParams) {
+        println!("{:?}", l)
+    }
+}
+
+async fn get_cached_server_binary(
+    container_dir: PathBuf,
+    node: &dyn NodeRuntime,
+) -> Option<LanguageServerBinary> {
+    (|| async move {
```
Comment (maxdeviant) [inline crates/zed/src/languages/purescript.rs:166]: We have an `async_maybe!` macro in our `util` crate that can make this a bit nicer than having the IIFE.

Here's an example of it used in another LSP adapter: https://github.com/zed-industries/zed/blob/5b9cc2619491c33b133254770ec2494010c2d016/crates/zed/src/languages/gleam.rs#L104

**Comment by maxdeviant** [comment]
I'm seeing some weird output in the generated `assets/licenses.md` file after running `script/generate-licenses`.

Namely, it seems we're pointing to the `tree-sitter-php` repo:

```
* [tree-sitter-purescript 1.0.0]( https://github.com/tree-sitter/tree-sitter-php )
```

It looks like the cause is that `Cargo.toml` is pointing to the wrong repository: https://github.com/ivanmoreau/tree-sitter-purescript/blob/4f623d7d245e76f8ab2a5ea663d8a257d48fbd97/Cargo.toml#L7

**Comment by ivanmoreau** [comment]
Hi @maxdeviant!

I solved the mentioned problems, including the removal of some code that wasn't essential for the functionality (it was merely a leftover from an existing implementation with Node that I used as a template).

**Comment by maxdeviant** [comment]
Looks good! Thanks for getting this added, @ivanmoreau!

**Pull request merged**